great_expectations/rule_based_profiler/parameter_builder/metric_multi_batch_parameter_builder.py

```diff
@@ -68,6 +70,8 @@ def __init__(
         self._metric_domain_kwargs = metric_domain_kwargs
         self._metric_value_kwargs = metric_value_kwargs
 
+        self._single_batch_mode = single_batch_mode
```

Comment:
If we allow for `str` input, can we cast it to a `bool` here so we don't have to handle both cases internal to this class?

Reply:
@billdirks this is for `$variable` syntax -- we do that throughout.  Thanks!

Reply:
For input from a file it makes sense to allow a `str`. But once it's in python why not convert to a `bool`. There is not string logic in code that I see.

Reply:
Actually when does the `$variable` syntax get resolved. If this `str` is anything but `""` it will resolve to `True`. To be more concrete why can't you do a `self._single_batch_mode = True if single_batch_mode else False`. 
Also, I may have missed this, but is there a test where you use a string and set this to `False`?

Reply:
I spoke with @alexsherstinsky and the string gets resolved on line 130. I think this could get resolved here or in the `single_batch_mode` property getter  since our logic all depends on the resolved value and now each caller has to resolve this themselves. @alexsherstinsky is going to file a followup ticket to resolve this since he wants to do this change for all the properties and not just this one.